lib/util/baseValidationError.ts

```diff
@@ -3,16 +3,12 @@
 
 import { Severity } from "./severity"
 import { ValidationResultSource } from "./validationResultSource"
+import { NodeError } from './validationError';
 
-export interface SerializedError {
-  readonly code: string
-  similarPaths: string[]
-}
-
-export interface BaseValidationError {
+export interface BaseValidationError<T extends NodeError<T>> {
   severity?: Severity
   errorCode?: string
-  errorDetails?: string | SerializedError
+  errorDetails?: T
```

Comment:
since we are only chanigng the types I guess the services wont need any changes ? maybe tooling if we rely on the type, please update those also and check

Reply:
Sure, I will.